docs/examples/vector_stores/AstraDBIndexDemo.ipynb

```diff
@@ -0,0 +1,261 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Astra DB\n",
+    "\n",
+    ">[Astra DB](https://astra.datastax.com) is a NoSQL, row-oriented, highly scalable and highly available database.\n",
+    "\n",
+    "To run this notebook you need a DataStax Astra DB instance running in the cloud (you can get one for free at [datastax.com](https://astra.datastax.com)).\n",
+    "\n",
+    "You should ensure you have `astrapy` installed:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install astrapy>=0.5.3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import needed packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import openai\n",
+    "\n",
+    "from llama_index import (\n",
+    "    VectorStoreIndex,\n",
+    "    SimpleDirectoryReader,\n",
+    "    StorageContext,\n",
+    ")\n",
+    "from llama_index.vector_stores import AstraDBVectorStore"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Please provide database connection parameters and secrets:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api_endpoint = input(\n",
+    "    \"\\nPlease enter your Database Endpoint URL (e.g. '0123abcd...'):\"\n",
+    ")\n",
+    "\n",
+    "token = getpass.getpass(\n",
+    "    \"\\nPlease enter your 'Database Administrator' Token (e.g. 'AstraCS:...'):\"\n",
+    ")\n",
+    "\n",
+    "OPENAI_API_KEY = getpass.getpass(\"OpenAI API Key:\")\n",
+    "openai.api_key = OPENAI_API_KEY"
```

Comment:
If you want to shave one line off, you should be able to replace these two with `os.environ["OPENAI_API_KEY"] = getpass("OPENAI_API_KEY = ")`